File: examples/review_diff_decision.py
from review_diff_forma import ReviewDiffOutput


def decide_review(output: ReviewDiffOutput) -> dict[str, object]:
    return {
        "status": "approve" if output.clean and len(output.findings) == 0 else "request_changes",
        "finding_count": len(output.findings),
        "paths": _unique_paths([finding.path for finding in output.findings]),
        "summary": output.summary,
    }


def _unique_paths(paths: list[str]) -> list[str]:
    return list(dict.fromkeys(paths))

File: examples/review_diff_decision.ts
import type { ReviewDiffOutput } from "./review_diff.forma.js";

export type ReviewDecisionStatus = "approve" | "request_changes";

export interface ReviewDecision {
  status: ReviewDecisionStatus;
  findingCount: number;
  paths: string[];
  summary: string;
}

export function decideReview(output: ReviewDiffOutput): ReviewDecision {
  return {
    status: output.clean && output.findings.length === 0 ? "approve" : "request_changes",
    findingCount: output.findings.length,
    paths: uniquePaths(output.findings.map((finding) => finding.path)),
    summary: output.summary,
  };
}

function uniquePaths(paths: string[]): string[] {
  return [...new Set(paths)];
}
